update-cache/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const UpdateCacheUrlFactory = require('./lib/UpdateCacheUrlFactory');
+const Signature = require('./lib/Signature');
+const Caches = require('amp-toolbox-cache-list');
+
+/**
+ * Creates an instance of UpdateCacheUrlFactory that uses the privateKey
+ * to sign the Urls.
+ *
+ * @param {string} privateKey Private Key to be used when signing Urls.
+ * @returns {UpdateCacheUrlFactory} an instance of UpdateCacheUrlFactory.
+ */
+function createUpdateCacheUrlFactory(privateKey) {
```

Comment:
Can we find a better name? A create method that creates a factory (which itself usually has a create method sounds wrong). 

How about 'createCacheUpdater(...)' and rename UpdateCacheUrlFactory to CacheUpdater?

Reply:
The purpose of this method is simply to make the API simpler to use.

My first idea was creating a static `createFactory` on `UpdateCacheUrlFactory`, but I'd end up introducing a dependency for `Caches` and `Signature` to it. So, I decided moving this to `index.js`.

I've seem a few instance of Factories of Factories or a mix of Factory / Builders. Maybe it would make sense to rename `createFactory` to `buildFactory`?

`CacheUpdater` is not a good name - this class does not update the cache. It only creates the URLs that do it.

Reply:
Hm, I'd prefer a static method in this case to avoid the double factory pattern. I really don't like the name Factory here, because it usually means a different thing (something that creates objects). You're right about CacheUpdater not being a good name though... 

How about:

```
class CacheUpdateUrlProvider {
 
  static create(privateKey) {
     const signature = new Signature(privateKey);
     const caches = new Caches();
     return new CacheUpdateUrlProvider(signature, caches);
  }

  constructor(..) {
  ...
}
```

I don't think it's a problem to have a dependency to Caches and Signature here, you have it anyway and if it is only at runtime.

Reply:
done